modules/sdk/src/main/scala/org/tessellation/sdk/domain/healthcheck/consensus/HealthCheckConsensus.scala

```diff
@@ -66,6 +90,23 @@ abstract class HealthCheckConsensus[
       if (inProgress.nonEmpty) manageRounds(inProgress) else Applicative[F].unit
     }
 
+  private def manageMissingProposals(
+    missingFromPeers: Set[PeerId],
+    key: K,
+    round: HealthCheckConsensusRound[F, K, A, B, C]
+  ): F[Unit] =
+    round.getRoundIds.flatMap { roundIds =>
+      Applicative[F].whenA(missingFromPeers.size > 0) {
+        logger.debug(
+          s"Missing proposals for round ids: ${roundIds} for key: ${key} are from peers: ${missingFromPeers}. Requesting proposals"
+        )
+      }
+    } >>
+      missingFromPeers.toList.traverse { requestProposal(_, round.getOwnRoundId) }
```

Comment:
I think that still peers can present other peer proposals as their own and we wouldn't know it. Probably there has to be a similar check as it is in rumor handlers.